pinot-core/src/main/java/org/apache/pinot/core/query/reduce/ExecutionStatsAggregator.java

```diff
@@ -70,10 +71,13 @@ public ExecutionStatsAggregator(boolean enableTrace) {
     _enableTrace = enableTrace;
   }
 
-  public void aggregate(ServerRoutingInstance routingInstance, DataTable dataTable) {
-    TableType tableType = routingInstance.getTableType();
-    String instanceName = routingInstance.getShortName();
+  public void aggregate(ServerRoutingInstance serverRoutingInstance, DataTable dataTable) {
+    String instanceName = serverRoutingInstance.getShortName();
     Map<String, String> metadata = dataTable.getMetadata();
+    TableType tableType = null;
+    if (metadata.get(DataTable.MetadataKey.TABLE.getName()) != null) {
+      tableType = TableNameBuilder.getTableTypeFromTableName(metadata.get(DataTable.MetadataKey.TABLE.getName()));
```

Comment:
@jackjlli this is where we're still able to capture metrics like number of distinct servers as well as per-table-type metrics.  The table name is added to DataTable metadata by the server via this PR, which enables disambiguating metrics.